domain/externalcontroller/state/state.go

```diff
@@ -77,6 +77,57 @@ WHERE  ctrl.uuid=$M.id`
 	return &ci, nil
 }
 
+func (st *State) ControllerForModel(
+	ctx context.Context,
+	modelUUID string,
+) (*crossmodel.ControllerInfo, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	q := `
+SELECT 	(ctrl.uuid, alias, ca_cert, address) as &ExternalController.* 
+FROM   	external_controller AS ctrl	
+	LEFT JOIN external_model AS model
```

Comment:
This should be just (inner) "JOIN".

Reply:
yes of course, just copied without thinking much, fixed